txscript/script.go

```diff
@@ -69,6 +69,87 @@ func IsPayToScriptHash(script []byte) bool {
 	return isScriptHash(pops)
 }
 
+// isWitnessScriptHash returns true if the passed script is a
+// pay-to-witness-script-hash transaction, false otherwise.
+func isWitnessScriptHash(pops []parsedOpcode) bool {
```

Comment:
Also, these should really only be true depending on the witness script version which I don't see anything for properly detecting and handling that.  I suspect this function, and the others in the same vain, should be taking the witness version and checking it against 0.

The BIP says:

```
If the version byte is 0, and the witness program is 32 bytes:
```